### PR TITLE
Rename argument to reduce confusion if attempting to use regular strings with the old -k

### PR DESCRIPTION
**Pull request by gfs**
This PR changes file path exclusion argument is now `-g` or `--exclusion-globs`.

This will reduce confusion for users coming from 1.2 where `-k` meant substring matches.

Also bring argument parity to the TagDiff command.

This is a semver relevant change so this bumps to `1.4`.

### CODE REVIEW

File: AppInspector.CLI/CLICmdOptions.cs [hunk 1]
```diff
@@ -48,22 +48,22 @@ public class CLIAnalyzeCmdOptions : CLICommandOptions
         [Option('e', "text-format", Required = false, HelpText = "Match text format specifiers", Default = "Tag:%T,Rule:%N,Ruleid:%R,Confidence:%X,File:%F,Sourcetype:%t,Line:%L,Sample:%m")]
         public string TextOutputFormat { get; set; } = "Tag:%T,Rule:%N,Ruleid:%R,Confidence:%X,File:%F,Sourcetype:%t,Line:%L,Sample:%m";
 
-        [Option("file-timeout", Required = false, HelpText = "If set, maximum amount of time in milliseconds to allow for processing each file.", Default = 60000)]
+        [Option('F',"file-timeout", Required = false, HelpText = "If non-zero, maximum amount of time in milliseconds to allow for processing each file. Default: 60000.", Default = 60000)]
```
Comment (scovetta) [inline AppInspector.CLI/CLICmdOptions.cs:51]: Does 0 mean 'infinity'?

Comment (gfs) [reply AppInspector.CLI/CLICmdOptions.cs:51]: Yes. Will add.

File: README.md [hunk 2]
```diff
@@ -9,10 +9,27 @@ Application Inspector is different from traditional static analysis tools in tha
...
+See more in the [wiki](https://github.com/microsoft/ApplicationInspector/wiki/7.-NuGet-Support)
+
+### Platform Dependent Binary
+- Download Application Inspector by selecting the pre-built package for the operating system of choice shown under the Assets section
+of the [Releases](https://github.com/microsoft/ApplicationInspector/releases).
+
+## Run Application Inspector
+
+- Nuget Tool: `appinspector analyze -s path/to/src`.
+- Platform Specific: `applicationinspector.cli.exe analyze -s path/to/src`
```
Comment (scovetta) [inline README.md:31]: These are supposed to be different, right? appinspector vs. applicationinspector.cli.exe?

Comment (gfs) [reply README.md:31]: Yes.

The nuget tool command is `appinspector`

The raw CLI binary is named `applicationinspector.cli` with `.exe` on windows.

**Review by scovetta — APPROVED** [review]
LGTM

**Review by jacobmsft — COMMENTED** [review]
LGTM

**Pull request merged**